sdk/keyvault/azure-security-keyvault-keys/src/private/key_constants.hpp

```diff
@@ -10,6 +10,8 @@
 #pragma once
 
 namespace Azure { namespace Security { namespace KeyVault { namespace Keys { namespace _detail {
+  constexpr static const char KeyVaultServicePackageName[] = "keyvault-keys";
```

Comment:
@LarryOsterman, all the package are sending package names `like-this`, while Attestation sends it `like.this` - should we update Attestation?

Reply:
The guidelines for user-agent say "client library package name as it appears to the developer, replacing slashes with dashes and removing the Azure indicator". So they should be "storage-blobs-cpp", "security-keyvault-keys-cpp", or "security-attestation-cpp".

Reply:
In other words, *all* of our current user-agent names are out of spec, attestation is further out of spec than all the other sdks.

Reply:
I brought up the -cpp suffix with the other architects (because it means that the letters "c", "p", and "p" appear twice in the user-agent string) and their feedback was that if the name of the package contains the language, we should keep the package name intact.

Reply:
Those are some good points we need to follow-up on.

I am considering changing all user-agent strings to include package name verbatims to adhere to the letter of the current SDK guideline, OR to update the guideline and leave the names as is, as an **open and separate** discussion from this implementation bug fix.

The goal of **this** PR is to fix the implementation bug and be internally consistent with all the other C++ SDKs (specifically KeyVault) so that we get appropriate characterization.